ClassNotFoundException during REPL startup

### Description
*Type*: <Bug>
*Summary*: running `lein repl` in a freshly initialized repo throws ClassNotFoundException.

#### Expected behavior
Get to the REPL prompt without any hassle.

#### Actual behavior
We get the below error:
```
Compiling 4 source files to /path-torepo/target/base+system+user+dev/classes
Exception in thread "main" java.lang.ExceptionInInitializerError
        at clojure.main.<clinit>(main.java:20)
Caused by: java.lang.ClassNotFoundException: commiteth.eth.contracts.TokenReg, compiling:(commiteth/eth/token_registry.clj:1:1)
... and the stack trace follows.
```

### Solution
*Summary*: seems that `env/dev/clj/user.clj` is autoloaded by Clojure no matter what (see e.g. [this](https://github.com/technomancy/leiningen/issues/1787) and [this](https://www.reddit.com/r/Clojure/comments/5f2ctk/cant_compile_java_sources_with_lein_javac/dahrkw5/)). Afterwards it loads `token-registry` as one of its dependencies, which in turn imports generated Java classes.

Suggested resolution is to dynamically require `commiteth.core` in `user` ns (like suggested [here](https://groups.google.com/forum/#!topic/clojure/4ytDn6OHabI)).

--- a/env/dev/clj/user.clj
+++ b/env/dev/clj/user.clj
@@ -1,13 +1,14 @@
 (ns user
   (:require [mount.core :as mount]
-            [commiteth.figwheel :refer [start-fw stop-fw cljs]]
-            commiteth.core))
+            [commiteth.figwheel :refer [start-fw stop-fw cljs]]))
 
 (defn start []
-  (mount/start-without #'commiteth.core/repl-server))
+  (require 'commiteth.core)
+  (mount/start-without (ns-resolve 'commiteth.core 'repl-server)))
 
 (defn stop []
-  (mount/stop-except #'commiteth.core/repl-server))
+  (require 'commiteth.core)
+  (mount/stop-except (ns-resolve 'commiteth.core 'repl-server)))
 
 (defn restart []
   (stop)

--- a/project.clj
+++ b/project.clj
@@ -116,10 +116,6 @@
                    :uberjar-name   "commiteth.jar"
                    :source-paths   ["env/prod/clj"]
                    :resource-paths ["env/prod/resources"]}
-   ;; :precomp profile allows to compile classes from commiteth.eth.contracts
-   ;; namespace before compiling clojure code. Otherwise ClassNotFound exception
-   ;; will be thrown
-   :precomp {:target-path "target/base+system+user+dev/" }
    :dev   {:dependencies   [[prone "1.1.4"]
                             [ring/ring-mock "0.3.1"]
                             [ring/ring-devel "1.6.2"]
@@ -146,7 +142,7 @@
                :optimizations :none
                :pretty-print  true}}]}
 
-           :prep-tasks     ["build-contracts" ["with-profile" "precomp" "javac"]]
+           :prep-tasks     ["build-contracts" "javac"]
            :doo            {:build "test"}
            :source-paths   ["env/dev/clj" "test/clj"]
            :resource-paths ["env/dev/resources"]

